src/install.c

```diff
@@ -44,16 +44,25 @@ static void install_args_update(RaucInstallArgs *args, const gchar *msg)
 
 static gchar *resolve_loop_device(const gchar *devicepath, GError **error)
 {
-	g_autofree gchar *devicename = NULL;
+	g_autoptr(GString) devicename = NULL;
+	g_autofree gchar *basename = NULL;
 	g_autofree gchar *syspath = NULL;
 	gchar *content = NULL;
 	GError *ierror = NULL;
 
 	if (!g_str_has_prefix(devicepath, "/dev/loop"))
 		return g_strdup(devicepath);
 
-	devicename = g_path_get_basename(devicepath);
-	syspath = g_build_filename("/sys/block", devicename, "loop/backing_file", NULL);
+	basename = g_path_get_basename(devicepath);
+
+	devicename = g_string_new(basename);
+	
+	/* Cut of any partition information like p1 */
+	devicename = g_string_set_size(devicename, sizeof("loop0"));
```

Comment:
Just had another quick look into alternatives, maybe a simple `g_strndup()` is less overhead here.

Reply:
The simple approach with cutting of a fixed number of characters will fail if more than 10 loop devices exist.

We need to check if it is a partition on a loop device, and in that case remove the `p\d+` part.